kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/model/VirtualCluster.java

```diff
@@ -180,22 +183,41 @@ public Optional<SslContext> getUpstreamSslContext() {
     private Optional<SslContext> buildDownstreamSslContext() {
         return tls.map(tlsConfiguration -> {
             try {
-                return Optional.of(tlsConfiguration.key()).map(NettyKeyProvider::new).map(NettyKeyProvider::forServer).orElseThrow().build();
+                var sslContextBuilder = Optional.of(tlsConfiguration.key()).map(NettyKeyProvider::new).map(NettyKeyProvider::forServer)
+                        .orElseThrow();
+
+                return configureTrustProvider(tlsConfiguration).apply(sslContextBuilder.clientAuth(toNettyClientAuth(tlsConfiguration)))
+                        .build();
             }
             catch (SSLException e) {
                 throw new UncheckedIOException(e);
             }
         });
     }
 
+    @NonNull
+    private static ClientAuth toNettyClientAuth(Tls tlsConfiguration) {
+        switch (tlsConfiguration.clientAuth()) {
+            case REQUIRED -> {
+                return ClientAuth.REQUIRE;
+            }
+            case REQUESTED -> {
+                return ClientAuth.OPTIONAL;
+            }
+            case NONE -> {
+                return ClientAuth.NONE;
+            }
+            default -> throw new IllegalStateException("Unexpected value: " + tlsConfiguration.clientAuth());
+        }
```

Comment:
Java 17's yield exhaustiveness means we can turn the runtime error into a compile time one.

```suggestion
    private static ClientAuth toNettyClientAuth(Tls tlsConfiguration) {
        return switch (tlsConfiguration.clientAuth()) {
            case REQUIRED -> ClientAuth.REQUIRE;
            case REQUESTED -> ClientAuth.OPTIONAL;
            case NONE -> ClientAuth.NONE;
        };
    }
```